hazelcast/src/test/java/com/hazelcast/cache/jsr/JsrTestUtil.java

```diff
@@ -17,19 +17,83 @@
 package com.hazelcast.cache.jsr;
 
 import com.hazelcast.core.Hazelcast;
+import com.hazelcast.instance.HazelcastInstanceFactory;
 
 import javax.cache.Caching;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+import static org.junit.Assert.fail;
 
 /**
- * Utility class responsible for setup/cleanup or member JSR tests.
+ * Utility class responsible for setup/cleanup of JSR member tests.
  */
 public final class JsrTestUtil {
 
     private JsrTestUtil() {
     }
 
+    public static void setup() {
+        setSystemProperties("server");
+    }
+
     public static void cleanup() {
-        Caching.getCachingProvider().close();
+        clearCachingProviderRegistry();
         Hazelcast.shutdownAll();
+        HazelcastInstanceFactory.terminateAll();
+    }
+
+    /**
+     * Sets the System properties for JSR related tests.
+     *
+     * @param providerType "server" or "client" according to your test type
+     */
+    public static void setSystemProperties(String providerType) {
+        /*
+        If we don't set this parameter the HazelcastCachingProvider will try to determine if it has to
+        create a client or server CachingProvider by looking for the client class. If you run the testsuite
+        from IDEA across all modules, that class is available, even though you might want to start a server
+        side test. This leads to a ClassCastException for server side tests, since a client CachingProvider
+        will be created. So we explicitly set this property to ease the test setups for IDEA environments.
+         */
+        System.setProperty("hazelcast.jcache.provider.type", providerType);
```

Comment:
I just use plain Strings here, since I don't want to trigger any classloading before the System properties are set. For example the usage of `GroupProperty.JCACHE_PROVIDER_TYPE.setSystemProperty(providerType)` already triggered the `MBeanServer` to get initialized without our custom implementation.